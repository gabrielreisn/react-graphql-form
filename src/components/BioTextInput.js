import * as React from 'react';
import { connect } from 'react-redux';
import TextField from 'material-ui/TextField';

import * as FormSelector from '../store/FormSelector';
import * as FormAction from '../store/FormAction';

const textFieldStyle = {
  rootElement: {
    width: '75%',
    marginBottom: '30px',
    marginTop: '20px',
  },
};

const BioTextInput = props => {
  const handleChange = event => {
    props.setYourBio(event.target.value);
  };

  return (
    <TextField
      hintText={props.hintText ? props.hintText.label : ''}
      fullWidth={true}
      style={textFieldStyle.rootElement}
      multiLine={props.multiLine}
      onChange={handleChange}
      value={props.your_bio}
    />
  );
};

const mapStateToProps = function(store, ownProps) {
  return {
    your_bio: FormSelector.getYourBio(store),
  };
};

const mapDispatchToProps = function(dispatch, ownProps) {
  return {
    setYourBio: param => {
      return dispatch(FormAction.setYourBio(param));
    },
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(BioTextInput);
