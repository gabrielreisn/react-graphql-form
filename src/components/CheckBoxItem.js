import * as React from 'react';
import { connect } from 'react-redux';
import Checkbox from 'material-ui/Checkbox';

import * as FormAction from '../store/FormAction';

class CheckBoxItem extends React.Component {
  state = {
    checked: false,
  };

  toggleCheck = () => {
    this.setState(prevState => {
      return {
        checked: !prevState.checked,
      };
    });

    const { label, setAdditionalExperience } = this.props;
    const { checked } = this.state;

    const checkBoxInfo = {
      label,
      value: checked,
    };

    setAdditionalExperience(checkBoxInfo);
  };

  render() {
    const { label } = this.props;
    const { checked } = this.state;
    return (
      <Checkbox label={label} checked={checked} onCheck={this.toggleCheck} />
    );
  }
}

const mapStateToProps = function(store, ownProps) {
  return {};
};

const mapDispatchToProps = function(dispatch, ownProps) {
  return {
    setAdditionalExperience: param => {
      return dispatch(FormAction.setAdditionalExperience(param));
    },
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(CheckBoxItem);
