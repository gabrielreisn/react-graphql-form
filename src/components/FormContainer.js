/* @flow */

import * as React from 'react';
import TextField from 'material-ui/TextField';
import DatePicker from 'material-ui/DatePicker';
import RaisedButton from 'material-ui/RaisedButton';
import Checkbox from 'material-ui/Checkbox';
import { RadioButton, RadioButtonGroup } from 'material-ui/RadioButton';
import SelectField from 'material-ui/SelectField';
import MenuItem from 'material-ui/MenuItem';

import styled from 'styled-components';

import { graphql } from 'react-apollo';
import gql from 'graphql-tag';

//using external inline-style to minimize re-render problems
const paperStyle = {
  width: '35%',
  height: '90vh',
  margin: 'auto 20px',
  backgroundColor: 'white',
  borderRadius: '8px',
  overflowY: 'auto',
};

const divWrapper = {
  margin: '20px 0',
};

const submitButtonStyle = {
  marginBottom: '40px',
  width: '75%',
};

const radioButtonGrouprStyle = {
  display: 'table',
  marginLeft: '12%',
  textAlign: 'left',
};

const textFieldStyle = {
  rootElement: {
    width: '75%',
    marginBottom: '30px',
    marginTop: '20px',
  },
};

const ItemWrapper = styled.div`
  display: flex;
  padding: 5vh;
`;

const TextHeaderWrapper = styled.div`
  color: #293e52;
  margin-left: 12%;
  text-align: left;
  font-weight: 600;
  margin-bottom: 4px;
`;

const LeftInfoBox = styled.div`
  width: 200px;
  height: 200px;
  background-color: white;
  border-radius: 8px;
  margin: 0 5px auto 20%;
`;

const InfoBoxHeader = styled.div`
  color: #0081ff;
  font-size: 22px;
  text-align: left;
  font-weight: bold;
  margin: 10px 10px 0 10px;
`;

const InfoBoxText = styled.div`
  color: #293e52;
  font-size: 22px;
  text-align: left;
  font-weight: bold;
  margin: 10px 10px 0 10px;
`;

type Props = {
  /* ... */
};

type State = {
  /* ... */
};

class TodoContainer extends React.Component<Props, State> {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    console.log('props data', this.props.javascript_library_of_choice);
    return (
      <ItemWrapper>
        <LeftInfoBox>
          <InfoBoxHeader>{this.props.organizationName ? this.props.organizationName : ''}</InfoBoxHeader>
          <InfoBoxText>{this.props.title ? this.props.title : ''}</InfoBoxText>
        </LeftInfoBox>

        <div style={paperStyle}>
          <div>
            <TextField
              hintText={this.props.your_name ? this.props.your_name.label : ''}
              fullWidth={true}
              style={textFieldStyle.rootElement}
            />
          </div>

          <div>
            <TextField
              hintText={this.props.your_bio ? this.props.your_bio.label : ''}
              fullWidth={true}
              style={textFieldStyle.rootElement}
              multiLine={true}
            />
          </div>
          <div>
            <SelectField
              hintText={this.props.primary_skill ? this.props.primary_skill.label : ''}
              style={textFieldStyle.rootElement}>
              {this.props.primary_skill
                ? this.props.primary_skill.options.map((data, i) => (
                  <MenuItem value={data} primaryText={data} key={i} />
                ))
                : ''}
            </SelectField>
          </div>
          <div style={divWrapper}>
            <TextHeaderWrapper>
              {this.props.javascript_library_of_choice ? this.props.javascript_library_of_choice.label : ''}
            </TextHeaderWrapper>

            <RadioButtonGroup name="shipSpeed" defaultSelected={'light'} style={radioButtonGrouprStyle}>
              <RadioButton value="light" label="React" />
              <RadioButton value="not_light" label="Angular" />
              <RadioButton value="ludicrous" label="Vue" />
              {/*
              {this.props.javascript_library_of_choice
                ? this.props.javascript_library_of_choice.options.map((data, i) => (
                    <RadioButton value={i} label={data} key={i} />
                  ))
                : ''}
                */}
            </RadioButtonGroup>
          </div>

          <div style={divWrapper}>
            <TextHeaderWrapper>
              {this.props.additional_experience ? this.props.additional_experience.label : ''}
            </TextHeaderWrapper>
            <div style={radioButtonGrouprStyle}>
              {this.props.additional_experience
                ? this.props.additional_experience.options.map((data, i) => (
                  <Checkbox label={data} key={i} checked={false} />
                ))
                : ''}
            </div>
          </div>
          <div>
            <DatePicker
              hintText={this.props.start_date ? this.props.start_date.label : ''}
              textFieldStyle={textFieldStyle.rootElement}
            />
          </div>
          <RaisedButton
            label={this.props.submitButtonText ? this.props.submitButtonText : 'submit'}
            backgroundColor={'#0081ff'}
            labelColor={'white'}
            style={submitButtonStyle}
          />
        </div>
      </ItemWrapper>
    );
  }
}

const PIPEFY_QUERY = gql`
  {
    publicForm(formId: "1lf_E0x4") {
      publicFormSettings {
        organizationName
        submitButtonText
        title
      }

      formFields {
        ... on ShortTextField {
          id
          label
        }
        ... on LongTextField {
          id
          label
        }
        ... on SelectField {
          id
          label
          options
        }
        ... on RadioVerticalField {
          id
          label
          options
        }
        ... on ChecklistVerticalField {
          id
          label
          options
        }
        ... on DateField {
          id
          label
        }
        __typename
      }
    }
  }
`;

export default graphql(PIPEFY_QUERY, {
  props: ({ data }) => {
    if (data.loading) {
      return {
        loading: data.loading,
      };
    }

    if (data.error) {
      return {
        error: data.error,
      };
    }

    return {
      loading: false,

      formFields: data.publicForm.formFields,
      publicFormSettings: data.publicForm.publicFormSettings,

      organizationName: data.publicForm.publicFormSettings.organizationName,
      submitButtonText: data.publicForm.publicFormSettings.submitButtonText,
      title: data.publicForm.publicFormSettings.title,

      your_name: data.publicForm.formFields[0],
      your_bio: data.publicForm.formFields[1],
      primary_skill: data.publicForm.formFields[2],
      javascript_library_of_choice: data.publicForm.formFields[3],
      additional_experience: data.publicForm.formFields[4],
      start_date: data.publicForm.formFields[5],
    };
  },
})(TodoContainer);
