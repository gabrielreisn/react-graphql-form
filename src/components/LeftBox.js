import * as React from 'react';
import styled from 'styled-components';

const LeftInfoBox = styled.div`
  width: 200px;
  height: 200px;
  background-color: white;
  border-radius: 8px;
  margin: 0 5px auto 20%;
`;

const InfoBoxHeader = styled.div`
  color: #0081ff;
  font-size: 22px;
  text-align: left;
  font-weight: bold;
  margin: 10px 10px 0 10px;
`;

const InfoBoxText = styled.div`
  color: #293e52;
  font-size: 22px;
  text-align: left;
  font-weight: bold;
  margin: 10px 10px 0 10px;
`;


class LeftBox extends React.Component {
    
    constructor(props) {
        super(props);
        this.state = {};
    }
  
    render () {
        return(
            <LeftInfoBox>
                <InfoBoxHeader>{this.props.organizationName ? this.props.organizationName : ''}</InfoBoxHeader>
                <InfoBoxText>{this.props.title ? this.props.title : ''}</InfoBoxText>
            </LeftInfoBox>
        );
    }
}

export default LeftBox;