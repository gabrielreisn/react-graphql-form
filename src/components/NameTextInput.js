import * as React from 'react';
import {connect} from 'react-redux';
import TextField from 'material-ui/TextField';

import * as FormSelector from '../selectors/FormSelector';
import * as FormAction from '../actions/FormAction';

const textFieldStyle = {
  rootElement: {
    width: '75%',
    marginBottom: '30px',
    marginTop: '20px',
  },
};

class NameTextInput extends React.Component {
  constructor(props) {
    super(props);

    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    this.props.setYourName(event.target.value);
  }

  render() {
    return (
      <div>
        <TextField
          hintText={this.props.hintText ? this.props.hintText.label : ''}
          fullWidth={true}
          style={textFieldStyle.rootElement}
          multiLine={this.props.multiLine}
          onChange={this.handleChange}
          value={this.props.your_name}
        />
      </div>
    );
  }
}

const mapStateToProps = function(store, ownProps) {
  return {
    your_name: FormSelector.getYourName(store),
  };
};

const mapDispatchToProps = function(dispatch, ownProps) {
  return {
    setYourName: param => {
      return dispatch(FormAction.setYourName(param));
    },
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(NameTextInput);
