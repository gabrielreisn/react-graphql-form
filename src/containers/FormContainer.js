// @flow

import * as React from 'react';
import { Query, Mutation } from 'react-apollo';
import CircularProgress from 'material-ui/CircularProgress';
import FormStoreContainer from './FormStoreContainer';

import { PIPEFY_MUTATION } from '../graphql/pipefyMutation';
import { PIPEFY_QUERY } from '../graphql/pipefyQuery';

const normalizeFields = fields => {
  return fields.reduce((accumulator, currentValue) => {
    accumulator[currentValue.__typename] = currentValue;
    return accumulator;
  }, {});
};

const FormContainer = () => (
  <Mutation mutation={PIPEFY_MUTATION}>
    {pipefyMutation => (
      <Query query={PIPEFY_QUERY} context={{ public: true }}>
        {({ loading, error, data }) => {
          if (loading) return <CircularProgress size={80} thickness={5} />;
          if (error) return alert(`Error!: ${error}`);

          const { publicForm: { publicFormSettings, formFields } } = data;
          return (
            <FormStoreContainer
              formMutation={pipefyMutation}
              publicFormSettings={publicFormSettings}
              formFields={normalizeFields(formFields)}
            />
          );
        }}
      </Query>
    )}
  </Mutation>
);

export default FormContainer;
