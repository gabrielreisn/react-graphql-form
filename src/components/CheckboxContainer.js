import * as React from 'react';
import CheckBoxItem from './CheckBoxItem';

import styled from 'styled-components';

const RadioButtonGrouprStyle = styled.div`
  display: table;
  margin-left: 12%;
  text-align: left;
`;

const DivWrapper = styled.div`
  margin: '20px 0';
`;

const TextHeaderWrapper = styled.div`
  color: #293e52;
  margin-left: 12%;
  text-align: left;
  font-weight: 600;
  margin-bottom: 4px;
`;

const CheckboxContainer = ({
  updateArrayField,
  checklistField: { options, label },
}) => (
  <DivWrapper>
    <TextHeaderWrapper>{label}</TextHeaderWrapper>
    <RadioButtonGrouprStyle>
      {options.map((data, i) => (
        <CheckBoxItem
          label={data}
          key={i}
          updateArrayField={updateArrayField}
        />
      ))}
    </RadioButtonGrouprStyle>
  </DivWrapper>
);

export default CheckboxContainer;
