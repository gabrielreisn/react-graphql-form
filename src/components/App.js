import React, {Component} from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import FormContainer from '../components/FormContainer';

import styled from 'styled-components';

const AppWrapper = styled.div`
  text-align: center;
  background-color: #e6ecf5;
  height: 100vh;
`;

const muiTheme = getMuiTheme({
  palette: {
    primary1Color: '#0081ff',
    primary2Color: '#0081ff',
    pickerHeaderColor: '#0081ff',
  },
});

class App extends Component {
  render() {
    return (
      <AppWrapper>
        <MuiThemeProvider muiTheme={muiTheme}>
          <FormContainer />
        </MuiThemeProvider>
      </AppWrapper>
    );
  }
}

export default App;
