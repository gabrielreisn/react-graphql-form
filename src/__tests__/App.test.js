import React from 'react';
import ReactDOM from 'react-dom';

import { ApolloProvider } from 'react-apollo';
import { ApolloClient } from 'apollo-client';

import App from '../components/App';
import LeftBox from '../components/LeftBox';
import { BioTextInput } from '../components/BioTextInput';
import { NameTextInput } from '../components/NameTextInput';
import { PIPEFY_QUERY, FormContainer } from '../components/FormContainer';

import { shallow } from 'enzyme';


describe('App component', () => {
  it('should render <App /> component', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toHaveLength(1);
  });
});

describe('LeftBox component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<LeftBox organizationName={'Pipefy Recruitment Test'} title={'Recruitment Survey'} />);

  })

  it('should render <LeftBox /> component', () => {

    expect(wrapper).toHaveLength(1);
  });
  it('should contains component props', () => {
    expect(wrapper.instance().props.organizationName).toBe('Pipefy Recruitment Test')
    expect(wrapper.instance().props.title).toBe('Recruitment Survey')

  });
  it('should match snapshot', () => {

    expect(wrapper).toHaveLength(1);
  });
});

describe('BioTextInput component', () => {
  it('should render <BioTextInput /> component with props', () => {
    const wrapper = shallow(<BioTextInput hintText={'your bio'} />);
    expect(wrapper.instance().props.hintText).toBe('your bio')
    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});

describe('NameTextInput component', () => {
  it('should render <NameTextInput /> component with props', () => {
    const wrapper = shallow(<NameTextInput hintText={'your name'} />);
    expect(wrapper.instance().props.hintText).toBe('your name')
    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});




