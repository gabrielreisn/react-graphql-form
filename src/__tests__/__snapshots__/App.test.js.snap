// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BioTextInput component should render <BioTextInput /> component with props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LeftBox
    hintText="your bio"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div />,
        <styled.div />,
      ],
      "id": "left-box",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div />,
          <styled.div />,
        ],
        "id": "left-box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
